src/blockchain/genesis_config.cpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "blockchain/genesis_config.hpp"
+
+#include <fmt/chrono.h>
+
+#include "app/configuration.hpp"
+#include "log/logger.hpp"
+#include "modules/production/read_config_yaml.hpp"
+
+namespace lean {
+  GenesisConfig::GenesisConfig(const log::LoggingSystem &logsys,
+                               const app::Configuration &app_config)
+      : config{readConfigYaml(app_config.genesisConfigPath()).value()} {
```

Comment:
Does `readConfigYaml` not generate any error? Errors should be caught and processed here if any.